### PR TITLE
Implement deprecation service class.

### PR DESCRIPTION
**Pull request by smarlowucf**
Sorry this is a large PR :thinking: .

### CODE REVIEW

File: mash/services/deprecation/service.py [hunk 1]
```diff
@@ -34,3 +45,371 @@ def post_init(self):
...
+                ),
+                extra=metata
+            )
+
+        # Don't send failure messages for always jobs.
+        if job.utctime != 'always' or job.status == SUCCESS:
+            self._publish_message(job)
+        job.listener_msg.ack()
+
+    def _publish_message(self, job):
```
[the remark this answers is not in the document]
Comment (smarlowucf) [reply mash/services/deprecation/service.py]: Yeah, it just publishes status messages to the next service which is PINT in our pipeline.

File: mash/services/deprecation/service.py [hunk 2]
```diff
@@ -34,3 +45,371 @@ def post_init(self):
...
+
+        Handle exceptions and errors that occur during deprecation and
+        logs info to job log.
+        """
+        job_id = event.job_id
+        job = self.jobs[job_id]
+        metata = job.get_metadata()
+
+        if job.utctime != 'always':
+            self._delete_job(job_id)
```
[the remark this answers is not in the document]
Comment (smarlowucf) [reply mash/services/deprecation/service.py]: I definitely overlooked this consideration. For always jobs we would stop at testing?

If so I think this should be handled as validation by job creator. And always job handling would need to be cleaned up from replication, publisher and deprecation.

[the remark this answers is not in the document]
Comment (rjschwei) [reply mash/services/deprecation/service.py]: Well on second thought there maybe people using mash that might want to always publish certain images. So the publish cannot be always is a "special sauce" for us kind of thing. We might consider adding some configuration option in the future.

[the remark this answers is not in the document]
Comment (smarlowucf) [reply mash/services/deprecation/service.py]: Okay, will not making any changes then.

**Pull request merged**